lib/kubernetes-deploy/kubernetes_resource/genericresource.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class GenericResource < KubernetesResource
```

Comment:
I've extracted this base class that represents a generic resource. This code is shared by several existing resources.

Reply:
Hm, `KubernetesResource` (the superclass) is already kinda "generic resource", no? Should we simply change the defaults in that class to what you have here? Regardless, please extract this change to a separate PR. 😄